File: lib/enums/user_role.dart
enum UserRole {
  student('Étudiant'),
  user('Utilisateur'),
  teacher('Enseignant'),
  admin('Administrateur');

  final String label;
  const UserRole(this.label);

  static UserRole fromString(String value) {
    switch (value.toLowerCase()) {
      case 'admin':
        return UserRole.admin;
      case 'teacher':
      case 'enseignant':
        return UserRole.teacher;
      case 'student':
      case 'étudiant':
        return UserRole.student;
      case 'user':
      default:
        return UserRole.user;
    }
  }

  @override
  String toString() => label;
}

File: app/Enums/UserRole.php
<?php

namespace App\Enums;

enum UserRole: string
{
    case ADMIN = 'admin';
    case TEACHER = 'teacher';
    case STUDENT = 'student';
    case USER = 'user';

    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }

    public static function toArray(): array
    {
        return [
            'admin' => 'Administrateur',
            'teacher' => 'Enseignant',
            'student' => 'Étudiant',
            'user' => 'Utilisateur',
        ];
    }
}